pilot/pkg/serviceregistry/kube/controller/serviceexportcontroller.go

```diff
@@ -0,0 +1,159 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+	"sigs.k8s.io/mcs-api/pkg/client/clientset/versioned"
+
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/queue"
+	"istio.io/pkg/log"
+)
+
+type ServiceExportController struct {
+	client        versioned.Interface
+	serviceClient corev1.CoreV1Interface
+
+	queue           queue.Instance
+	serviceInformer cache.SharedInformer
+
+	clusterLocalHosts []string //hosts marked as cluster-local, which will not have serviceeexports generated
+}
+
+func NewServiceExportController(kubeClient kube.Client, clusterLocalHosts []string) (*ServiceExportController, error) {
+	serviceExportController := &ServiceExportController{
+		client:            kubeClient.MCSApis(),
+		serviceClient:     kubeClient.Kube().CoreV1(),
+		queue:             queue.NewQueue(time.Second),
+		clusterLocalHosts: clusterLocalHosts,
+	}
+
+	serviceExportController.serviceInformer = kubeClient.KubeInformer().Core().V1().Services().Informer()
```

Comment:
Donot create a new informer, this will duplicate the cache

Reply:
Its using shared index informer under the hood, it should be shared